onadata/apps/api/viewsets/floip_viewset.py

```diff
@@ -131,15 +198,30 @@ def responses(self, request, uuid=None):
             else:
                 queryset = xform.instances.values_list('json', flat=True)
 
-            paginate_queryset = self.paginate_queryset(queryset)
-            if paginate_queryset:
-                data['attributes']['responses'] = floip_list(paginate_queryset)
-                response = self.get_paginated_response(data)
+            queryset = self.filter_queryset(queryset)
+            afterCursor = None
+            beforeCursor = None
+            if 'page[afterCursor]' in self.request.query_params:
+                afterCursor = int(
+                    self.request.query_params['page[afterCursor]'])
+
+            if 'page[beforeCursor]' in self.request.query_params:
+                beforeCursor = int(
+                    self.request.query_params['page[beforeCursor]']
+                )
+
+            if "page[size]" in self.request.query_params:
+                responses = convert_instances_to_floip_list(
```

Comment:
Added a new function that returns the FLOIP responses as a list instead of a generator in order to paginate the list. _We're not paginating the queryset since the actual items that need to be paginated are the `responses` which are just the questions answered within an Instance... Paginating the Instances would not work as the spec describes_